src/api/Shrooms.Domain.ServiceValidators/Validators/FilterPresets/IFilterPresetValidator.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Shrooms.Contracts.DataTransferObjects.FilterPresets;
+using Shrooms.Contracts.Enums;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Shrooms.Domain.ServiceValidators.Validators.FilterPresets
+{
+    public interface IFilterPresetValidator
+    {
+        void CheckIfFilterTypesAreValid(FilterType[] filterTypes);
+
+        void CheckIfFilterTypesContainDuplicates(FilterType[] filterTypes);
+
+        void CheckIfPageTypeExists(PageType page);
+
+        void CheckIfCountsAreEqual<T1, T2>(T1 firstCollection, T2 secondCollection)
+            where T1 : ICollection
+            where T2 : ICollection;
+
+        void CheckIfMoreThanOneDefaultPresetExists(ManageFilterPresetDto manageFilterPresetDto);
+
+        void CheckIfFilterPresetsContainsUniqueNames(IEnumerable<FilterPresetDto> filterPresetDtos);
```

Comment:
Didn't notice it at first - but it same issue with plural and "contains" (should "presets contain").